diff.py

```diff
@@ -965,10 +965,11 @@ def serialize_data_references(references: List[Tuple[int, int, str]]) -> str:
 def maybe_get_objdump_source_flags(config: Config) -> List[str]:
     flags = []
 
-    if config.show_line_numbers or config.source:
+    if config.show_line_numbers or config.show_source or True:
         flags.append("--line-numbers")
 
-    if config.source:
+    config.source_old_binutils = True
```

Comment:
Was this added for testing and supposed to be removed before committing? ;)

FYI, I hadn't tested source_old_binutils so far, but the lack of --source-comment seems to prevent asm-differ from distinguishing between empty source lines and empty objdump lines — which causes broken output with richer debug info that contains empty source lines. The latter show up as added lines in this branch.

Reply:
Aside from the mistakes I had previously included, is the behavior different from the version on `main`? 

For me, on binutils version `2.26.1` ("old"), `objdump` skips printing empty source lines.

Reply:
I had been testing with IDO for `mips`, and GCC for `aarch64`, with objdumps version 2.26 & 2.37.

I was able to reproduce this by not using the `diff_obj` mode: there is an earlier check that stripped empty lines, but only if `config.diff_obj` was true. I refactored that to apply regardless of mode. 

This now mostly fixes the problem that exists on `main` for `--source-old-binutils`. One minor drawback is that now with this flag, completely empty source lines will be excluded from the source view in `diff.py`'s output. (This is only with `--source-old-binutils`; blank lines are included with `--source`, so this seems like a reasonable tradeoff)

Reply:
Yeah that sounds reasonable since it's not like we can tell if the blank line is source code or not. (I wonder why there was a check for obj mode though 🤔)